### PR TITLE
man: update sctp.7 and sctp_sendmsg.3

### PR DESCRIPTION
**Pull request by lxin**
It's been a long time since the man doc was updated last time,
and there are a lot of futures having been added, and we need
a big update for these socket options, control messages and
notifications etc. in the man doc.

v1->v2:
  - improve some descriptions in man/sctp.7 according to Marcelo's feedback.

### CODE REVIEW

File: man/sctp.7 [hunk 1]
```diff
@@ -251,126 +251,176 @@ to read or
...
 This option is used to get or set the protocol parameters used to
-initialize and bound retransmission timeout(RTO). The structure sctp_rtoinfo
-defined in /usr/include/netinet/sctp.h is used to access and modify these
-parameters.
+initialize and bound retransmission timeout(RTO).
+
+The parameter type is struct sctp_rtoinfo: for write and read,
+srto_assoc_id is a specified assoc_id or SCTP_FUTURE_ASSOC.
+
+Default: srto_max=sysctl_net.sctp.rto_max, srto_min=sysctl_net.sctp.rto_max,
```
[the remark this answers is not in the document]
Comment (lxin) [reply man/sctp.7:387]: I wanted to tell users "sysctl net.sctp.rto_max" can provide the default value.
Do you think "net->sctp.rto_min" is more clear than "sysctl_net.sctp.rto_max" for that?

File: man/sctp.7 [hunk 2]
```diff
@@ -251,126 +251,176 @@ to read or
...
 This option is used to both examine and set various association and endpoint
-parameters. The structure sctp_assocparams defined in
-/usr/include/netinet/sctp.h is used to access and modify these parameters.
+parameters.
+
+The parameter type is struct sctp_assocparams: for write and read,
+sasoc_assoc_id is a specified assoc_id or SCTP_FUTURE_ASSOC.
+
+Default: sasoc_asocmaxrxt=sysctl_net.sctp.association_max_retrans,
+sasoc_cookie_life=sysctl_net.sctp.valid_cookie_life.
```
[the remark this answers is not in the document]
Comment (lxin) [reply man/sctp.7:398]: Other fields are not for writing, but for reading only, and those fields in asoc are decided during the connecting, and those in sp are not really used. Maybe I should say:

              The parameter type is struct sctp_assocparams, for write and read. sasoc_assoc_id is a specified assoc_id or SCTP_FUTURE_ASSOC. sasoc_peer_rwnd, sasoc_local_rwnd and sasoc_number_peer_destinations are not really used for writing.

File: man/sctp.7 [hunk 3]
```diff
@@ -251,126 +251,176 @@ to read or
...
-defined in /usr/include/netinet/sctp.h. The input parameters accepted by this
-call include sinfo_stream, sinfo_flags, sinfo_ppid, sinfo_context,
-sinfo_timetolive. The user must set the sinfo_assoc_id field to identify the
-association to affect if the caller is using the one-to-many style.
+inclusion of ancillary data. This option has been obsoleted by
+SCTP_DEFAULT_SNDINFO.
+
+The parameter type is struct sctp_sndrcvinfo: for write, sinfo_assoc_id
+is a specified assoc_id or SCTP_{FUTURE|CURRENT|ALL}_ASSOC; for read,
+sinfo_assoc_id is a specified assoc_id or SCTP_FUTURE_ASSOC.
```
[the remark this answers is not in the document]
Comment (lxin) [reply man/sctp.7:485]: 👍, thanks.

File: man/sctp.7 (hunk 2 above)
[the remark this answers is not in the document]
Comment (marceloleitner) [reply man/sctp.7:398]: The format of these is what is confusing me. So what about add a copy of the struct, and annotate which fields are R, W or RW?

[the remark this answers is not in the document]
Comment (lxin) [reply man/sctp.7:398]:
```
... are not really used. Note that some fields of this structure are for reading only:

  struct sctp_assocparams {
          sctp_assoc_t    sasoc_assoc_id;
          __u16           sasoc_asocmaxrxt; (RW)
          __u16           sasoc_number_peer_destinations; (R)
          __u32           sasoc_peer_rwnd; (R)
          __u32           sasoc_local_rwnd; (R)
          __u32           sasoc_cookie_life; (RW)
  };
```
sure, and this looks better.

**Comment by marceloleitner** [comment]
LGTM now

**Pull request merged**